compiler/luci/service/src/Nodes/CirclePad.test.cpp

```diff
@@ -31,3 +33,59 @@ TEST(CloneNodeTest, clone_Pad)
   auto cloned_pad = dynamic_cast<luci::CirclePad *>(cloned);
   ASSERT_NE(nullptr, cloned_pad);
 }
+
+TEST(ShapeRuleTest, pad_dynamic_shape)
+{
+  luci::CirclePad pad;
+  luci::CircleInput input;
+  luci::CircleConst padddings;
+
+  loco::TensorShape shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  input.shape({1, 2, 3, 4});
+  input.shape_status(luci::ShapeStatus::VALID);
+  input.dim(2).unset();
+
+  padddings.dtype(loco::DataType::S64);
+  padddings.shape({4, 2});
+  padddings.shape_status(luci::ShapeStatus::VALID);
+
+  const loco::DataType S64 = loco::DataType::S64;
+  uint32_t t = 64 * 8;
+  padddings.size<S64>(t);
+
+  pad.input(&input);
+  pad.paddings(&padddings);
+
+  ASSERT_TRUE(shape_inf_rule.infer(&pad, shape));
+  ASSERT_EQ(shape.rank(), 4);
+  ASSERT_TRUE(shape.dim(0).known());
+  ASSERT_TRUE(shape.dim(1).known());
+  ASSERT_FALSE(shape.dim(2).known());
+  ASSERT_TRUE(shape.dim(3).known());
+
+  ASSERT_EQ(1, shape.dim(0).value());
+  ASSERT_EQ(2, shape.dim(1).value());
+  ASSERT_EQ(0, shape.dim(2).value());
+  ASSERT_EQ(4, shape.dim(3).value());
+  pad.drop();
```

Comment:
Q) why is `drop()` called ?

Reply:
If drop() is not called, `./nncc test` fails with this message.
`luci_service_test: /home/ssafy/one/compiler/loco/src/IR/Node.cpp:28: virtual loco::Node::~Node(): Assertion '_uses.size() == 0' failed.`
I think in this test case, I made `CirclePad` uses `paddings` . When test ends and CirclePad is destoryed, `paddings` still remaining and cause an error.
So I explicitly drop pad node when test passes.

Reply:
@icodo98 
While writing the test code, it seems that the order of declarations is important.
If the CirclePad is declared first, a drop is required, but if the parameters used in CirclePad are declared first, then drop() is not necessary. like,
```c++
luci::CircleInput input;
luci::CirclePad pad;
```

Reply:
> but if the parameters used in CirclePad are declared first, then drop() is not necessary. like,
> 
> ```c++
> luci::CircleInput input;
> luci::CirclePad pad;
> ```

I didn't know that.
Although I already decided to use `graph`, I'll refer to that next time I write TC.